src/esp/physics/RigidObject.cpp

```diff
@@ -9,7 +9,9 @@ namespace esp {
 namespace physics {
 
 RigidObject::RigidObject(scene::SceneNode* rigidBodyNode)
-    : Magnum::SceneGraph::AbstractFeature3D(*rigidBodyNode) {}
+    : Magnum::SceneGraph::AbstractFeature3D(*rigidBodyNode) {
+  visualNode_ = &rigidBodyNode->createChild();
```

Comment:
if you use the initializer list, this could be a reference, unless we plan to have a rigidobject without a visual part though. Either way could you use the initializer list?